nemo/tron/converter/common.py

```diff
@@ -0,0 +1,346 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+import torch
+import torch.distributed as dist
+import yaml
+from megatron.core import parallel_state
+from megatron.core.dist_checkpointing.strategies.torch import TorchDistLoadShardedStrategy
+from megatron.core.optimizer import OptimizerConfig
+
+from nemo.collections.llm.gpt.model.base import GPTConfig, torch_dtype_from_mcore_config
+from nemo.collections.llm.t5.model.t5 import T5Config
+from nemo.tron.checkpointing import save_checkpoint
+from nemo.tron.config import CheckpointConfig, ConfigContainer, LoggerConfig, TokenizerConfig
+from nemo.tron.state import GlobalState
+from nemo.tron.tokenizers.tokenizer import _HuggingFaceTokenizer
+from nemo.tron.utils.instantiate_utils import instantiate
+
+if TYPE_CHECKING:
+    from transformers import AutoModelForCausalLM, AutoTokenizer, PretrainedConfig
+
+
+logger = logging.getLogger(__name__)
+
+HF_ASSETS_DIR = "hf_assets"
+
+
+@contextmanager
+def megatron_cpu_init_context(config):
+    """ """
+    _orig_use_cpu_initialization = config.use_cpu_initialization
+
+    config.use_cpu_initialization = True
+
+    yield
+
+    config.use_cpu_initialization = _orig_use_cpu_initialization
+
+
+@contextmanager
+def temporary_distributed_context():
+    if "MASTER_ADDR" not in os.environ:
+        os.environ["MASTER_ADDR"] = "localhost"
+    if "MASTER_PORT" not in os.environ:
+        os.environ["MASTER_PORT"] = "12355"
```

Comment:
if these were added as env vars, they should also be removed after control is yielded back

Reply:
Good point, will update.